jsonp-policy/src/main/java/io/apiman/plugins/jsonp_policy/JsonpPolicy.java

```diff
@@ -76,8 +86,12 @@ public void write(IApimanBuffer chunk) {
                     if (firstChunk) {
                         IApimanBuffer buffer = bufferFactory.
                                 createBuffer(callbackFunctionName.length() + OPEN_PARENTHESES.length());
-                        buffer.append(callbackFunctionName);
-                        buffer.append(OPEN_PARENTHESES);
+                        try {
+                            buffer.append(callbackFunctionName, encoding);
+                            buffer.append(OPEN_PARENTHESES, encoding);
+                        } catch (UnsupportedEncodingException e) {
+                            throw new RuntimeException(e);
```

Comment:
Darn that's a great question!  :)  Perhaps an exception *is* the proper way to go here (although presumably it should be a declared IOException).  That would require some changes to core.

@alexkieling - for now could you add a TODO comment to this line saying something about "replace in future with IOException"?